src/Report/EnabledMembers.php

```diff
@@ -0,0 +1,167 @@
+    return;
+}
+
+class EnabledMembers extends Report
+{
+    protected $title = 'Members with MFA enabled on their account';
+
+    protected $description = 'Identifies all members on this site, with columns added to indicate the status each has'
+        . ' in regards to multi factor authentication method registration. E.g. how many methods are set up by that'
+        . ' member, which method types, and who has opted not to set up MFA (if this option is enabled for the site)';
+
+    protected $dataClass = Member::class;
+
+    public function title()
+    {
+        return _t(__CLASS__ . '.TITLE', parent::title());
+    }
+
+    public function description()
+    {
+        return _t(__CLASS__ . '.DESCRIPTION', parent::description());
+    }
+
+    /**
+     * @param array $params
+     */
+    public function sourceRecords($params)
+    {
+        $methods = $this->getMethodClassToTitleMapping();
+
+        // Having a WHERE limits the results, which means the the COUNT will be off (GROUP BY applied after WHERE)
+        // So clip the count filter and apply it after the database query, only if Methods filter is also set
+        // this way we can delegate to the database where possible, which should be faster.
+        if (isset($params['Count'])) {
+            $desiredCount = (int)$params['Count'];
+            if (isset($params['Methods'])) {
+                $countFilter = $desiredCount;
+                unset($params['Count']);
+            } elseif ($desiredCount) {
+                $params['Count'] = $desiredCount++;
+            }
+        }
+
+        $members = $this->applyParams(Member::get(), $params)->toArray();
+        array_walk($members, function ($member) use ($methods) {
+            $defaultMethod = $member->getDefaultRegisteredMethod();
+            if ($defaultMethod) {
+                $defaultMethodClassName = $defaultMethod ? $defaultMethod->MethodClassName : null;
+                $backupClass = get_class(Injector::inst()->get(MethodRegistry::class)->getBackupMethod());
+                $registeredMethods = $member
+                    ->RegisteredMFAMethods()
+                    ->exclude('MethodClassName', $backupClass)
+                    ->column('MethodClassName');
+                $registeredMethodNames = array_map(function ($methodClass) use ($methods) {
+                    return $methods[$methodClass];
+                }, $registeredMethods);
+                $registeredMethodList = implode(', ', $registeredMethodNames);
+                $methodCount = count($registeredMethods);
+                $member->methodCount = $methodCount ?: '';
+                $member->registeredMethods = $registeredMethodList;
+                $member->defaultMethodName = $methods[$defaultMethodClassName] ?? '';
+            }
+            $member->skippedRegistration = $member->HasSkippedMFARegistration
+                ? $member->dbObject('HasSkippedMFARegistration')->Nice()
+                : '';
```

Comment:
I think this should only be filled if the user hasn't registered MFA. I have a user showing "number of registered methods" is one, but there's also "Yes" for skipped registration.

Reply:
This is useful information for statistical analysis on uptake.

Reply:
I just think it's confusing to show that a user has a method (ie. they've set up MFA) but they're also showing as having skipped MFA. I think it should remain blank (implying "N/A").

Reply:
@NightJar

Reply:
I agree with @NightJar here. Whether they've got registered methods and whether they've skipped registration are two mutually exclusive things.

That said, in a similar way to the previous comment I left, I don't think you should be giving it an empty value instead of "No".

Reply:
Now says "No".